src/utils/narrow.js

```diff
@@ -155,3 +155,12 @@ export const isSameNarrow = (narrow1: Narrow, narrow2: Narrow): boolean =>
   Array.isArray(narrow1) && Array.isArray(narrow2) && isEqual(narrow1, narrow2);
 
 export const parseNarrowString = (narrowStr: string): Narrow => JSON.parse(unescape(narrowStr));
+
+const starredNarrow: Narrow = [
+  {
+    operator: 'is',
+    operand: 'starred',
+  },
+];
+
+export const starredNarrowStr = JSON.stringify(starredNarrow);
```

Comment:
This can get merged as a separate commit, if needed.
Not sure if we want to do that though, as are currently generating this narrow via `specialNarrow`.

Reply:
If we want to keep it, maybe we would do something like `starredNarrow = specialNarrow('starred')` and then replace every use of `specialNarrow('starred')` with `starredNarrow`

Reply:
I'll update the commit to use `specialNarrow`.

Reply:
I can split the commit and update every use of `specialNarrow('starred')` to `starredNarrow`. For consistency, then, we should also have similar exports for every use of `specialNarrow`.

Should we use upper snake case to denote certain narrows that are constant? I.e. `HOME_NARROW`, `HOME_NARROW_STR`, `STARRED_NARROW`, etc.?

Reply:
@jackrzhang yeah that UPPER_CASE naming is a good idea. Thanks for noticing that. 
I did go through the source code and created a PR for that #2848

Reply:
Yeah, this sounds like a good plan -- replace the uses of `specialNarrow('starred')` with a new `STARRED_NARROW`, as a separate preparatory commit.

I just merged #2848 , so CONSTANT_CASE will blend in nicely.

Reply:
And I think doing so for the other special narrows -- `mentioned` and `private` are the ones I see -- would be good too. In fact we already have `ALL_PRIVATE_NARROW` for one of those.

Reply:
I've added two commits for `starred` and `mentioned`.